core/src/test/java/io/questdb/cutlass/line/udp/LineUdpPartitionReadOnlyTest.java

```diff
@@ -0,0 +1,286 @@
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.cutlass.line.udp;
+
+import io.questdb.Bootstrap;
+import io.questdb.ServerMain;
+import io.questdb.cairo.*;
+import io.questdb.cairo.security.AllowAllCairoSecurityContext;
+import io.questdb.cairo.vm.Vm;
+import io.questdb.cairo.vm.api.MemoryMARW;
+import io.questdb.cutlass.line.AbstractLinePartitionReadOnlyTest;
+import io.questdb.cutlass.line.LineTcpSender;
+import io.questdb.cutlass.line.LineUdpSender;
+import io.questdb.griffin.SqlCompiler;
+import io.questdb.griffin.SqlExecutionContext;
+import io.questdb.griffin.SqlExecutionContextImpl;
+import io.questdb.mp.SOCountDownLatch;
+import io.questdb.network.Net;
+import io.questdb.network.NetworkFacadeImpl;
+import io.questdb.std.*;
+import io.questdb.std.str.Path;
+import org.junit.*;
+
+import static io.questdb.cairo.TableUtils.createTable;
+import static io.questdb.test.tools.TestUtils.insertFromSelectPopulateTableStmt;
+import static io.questdb.test.tools.TestUtils.assertSql;
+import static io.questdb.test.tools.TestUtils.assertMemoryLeak;
+
+public class LineUdpPartitionReadOnlyTest extends AbstractLinePartitionReadOnlyTest {
+
+    @Test
+    public void testActivePartitionReadOnlyAndNoO3UDP() throws Exception {
+        Assume.assumeFalse(Os.isLinux()); // cannot debug this myself
+        String tableName = testName.getMethodName();
+        int numEvents = 5000;
+        SOCountDownLatch sendComplete = new SOCountDownLatch(1);
+        assertServerMainWithLineUDP(
+                tableName,
+                () -> {
+                    long[] timestampNano = {lastPartitionTs};
+                    try (LineUdpSender sender = new LineUdpSender(NetworkFacadeImpl.INSTANCE, 0, Net.parseIPv4("127.0.0.1"), ILP_PORT, 200, 1)) {
+                        for (int tick = 0; tick < numEvents; tick++) {
+                            int tickerId = 0;
+                            timestampNano[tickerId] += lineTsStep;
+                            sender.metric(tableName)
+                                    .tag("s", "lobster")
+                                    .field("l", 88)
+                                    .field("i", 2124)
+                                    .at(timestampNano[tickerId]);
+                        }
+                        sender.flush();
+                    } finally {
+                        sendComplete.countDown();
+                    }
+                },
+                sendComplete,
+                0,
+                TABLE_START_CONTENT,  // <-- read only, remains intact
+                false, false, false, true
+        );
+    }
+
+    @Test
+    public void testActivePartitionReadOnlyAndO3OverActivePartitionUDP() throws Exception {
+        Assume.assumeFalse(Os.isLinux()); // cannot debug this myself
```

Comment:
We shouldn't merge this PR until we fix the crashes. Do you need some help with fixing the code on Linux?

Reply:
hola @puzpuzpuz yes please, would you please be able to run the tests in a linux Box, a ptr must be offset, suspect `LinuxMMLineUdpReceiver`.

Reply:
I am setting up a linux box in my windows box

Reply:
these tests pass on my ubuntu with repeat until failure true and Tests passed == 5000.
the problem must be in CI, for some reason. Would you be able to tell where @puzpuzpuz ?

Reply:
Let me check if I can reproduce the failure locally.

Reply:
The segfault seems to be ARM specific. I'm not able to reproduce it on x86, but looks like arm64-linux build has crashed already:
```
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x0000ffffa47e9b70, pid=3035, tid=9182
#
# JRE version: OpenJDK Runtime Environment Corretto-11.0.17.8.1 (11.0.17+8) (build 11.0.17+8-LTS)
# Java VM: OpenJDK 64-Bit Server VM Corretto-11.0.17.8.1 (11.0.17+8-LTS, mixed mode, tiered, compressed oops, g1 gc, linux-aarch64)
# Problematic frame:
# J 6153 c2 sun.misc.Unsafe.getLong(J)J jdk.unsupported@11.0.17 (8 bytes) @ 0x0000ffffa47e9b70 [0x0000ffffa47e9b40+0x0000000000000030]
#
# No core dump will be written. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# An error report file with more information is saved as:
# /usr/share/vsts-agent/_work/1/s/core/hs_err_pid3035.log
Compiled method (c2)  280881 6153       4       sun.misc.Unsafe::getLong (8 bytes)
 total in heap  [0x0000ffffa47e9990,0x0000ffffa47e9c20] = 656
 relocation     [0x0000ffffa47e9b00,0x0000ffffa47e9b10] = 16
 main code      [0x0000ffffa47e9b40,0x0000ffffa47e9bc0] = 128
 stub code      [0x0000ffffa47e9bc0,0x0000ffffa47e9be0] = 32
 metadata       [0x0000ffffa47e9be0,0x0000ffffa47e9bf0] = 16
 scopes data    [0x0000ffffa47e9bf0,0x0000ffffa47e9bf8] = 8
 scopes pcs     [0x0000ffffa47e9bf8,0x0000ffffa47e9c18] = 32
 dependencies   [0x0000ffffa47e9c18,0x0000ffffa47e9c20] = 8
```

I'm afraid that you'd have to spin up an ARM EC2 instance and debug the problem there. Let me know if you need any help on that.

Reply:
I have disabled the tests for Linux arm, and I plan on not fixing.

Reply:
remains a challenge for a follow pr, I am not aware of users on UDP linux arm.

Reply:
Fixed with d1f59a78104b80ecc57807ec14dd045505c194e4